### PR TITLE
Deleting response objects

### LINKED ISSUE #118: Enable removing response objects from surveys
We would need to worry about attached links and delete them. We can use `CASCADE` in our queries to enable this deletion, but we should also warn users that we're about to delete _n_ relationships before we do.

### PR DESCRIPTION
**Pull request by rmarow**
resolves #118

### CODE REVIEW

File: usaon_vta_survey/templates/macros/javascript_delete.j2 [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {% macro add_javascript__delete() -%}
   <script>
-    function remove_relationship(url, response) {
+    function remove(url, response) {
```
Comment (MattF-NSIDC) [inline usaon_vta_survey/templates/macros/javascript_delete.j2:3]: I feel the JS function should match the macro name, e.g. call it `delete`, or change the macro to `add_javascript__remove`

Comment (rmarow) [reply usaon_vta_survey/templates/macros/javascript_delete.j2:3]: good call, it was doing a weird highlight when I tried calling the function `delete` so I figured it may have been some sort of keyword and didnt wanna mess with that.

File: usaon_vta_survey/templates/macros/javascript_remove.j2 [hunk 2]
```diff
@@ -0,0 +1,15 @@
+{% macro add_javascript__remove() -%}
+  <script>
+    function remove(url, response) {
+      fetch(url, { method: 'DELETE'})
+          .then(response => {
+              if (response.ok && response.redirected) {
+                    location.assign(response.url)
+              } else {
+                    console.error(`Something unexpected happened when trying to delete ${response.url}`)
```
Comment (MattF-NSIDC) [inline usaon_vta_survey/templates/macros/javascript_remove.j2:9]:
```suggestion
                    console.error(`Something unexpected happened with query: DELETE ${response.url}: ${response.body}`)
```

I'm not sure if `response.body` is correct, but whatever attribute holds the content would be great to log. You can find this in the network tab of browser dev tools, but if you didn't have it open already that info wouldn't have been recorded, so I think it's good to `console.error` it.

File: usaon_vta_survey/templates/response/applications.html [hunk 3]
```diff
@@ -2,7 +2,7 @@
 {% from 'bootstrap5/form.html' import render_form %}
 {% from 'bootstrap5/utils.html' import render_icon %}
 {% from 'macros/javascript_highcharts.j2' import add_javascript__highcharts %}
-{% from 'macros/javascript_delete.j2' import add_javascript__delete %}
+{% from 'macros/javascript_remove.j2' import add_javascript__remove %}
```
Comment (MattF-NSIDC) [inline usaon_vta_survey/templates/response/applications.html:5]: :sparkles:

File: usaon_vta_survey/templates/response/observing_systems.html [hunk 4]
```diff
@@ -14,7 +16,15 @@ <h3>Observing systems</h3>
...
     {% for observing_system in response.observing_systems %}
       <tr>
-        <td>{{observing_system.short_name}}</td>
+        <td>{{observing_system.short_name}}
+        <a href="#" onclick="remove('{{url_for(
+            'obs.delete_response_observing_system',
+            survey_id=survey.id,
+            response_observing_system_id=observing_system.id,
+                    )}}'
+                    )">{{render_icon('file-minus-fill', color='danger')}}
```
Comment (MattF-NSIDC) [inline usaon_vta_survey/templates/response/observing_systems.html:25]: Nitpick: the horizontal spacing here is a touch hard to follow. I'd suggest `prettier`, but it doesn't know how to Jinja. There is a plugin... https://github.com/davidodenwald/prettier-plugin-jinja-template

But it seems like the wrong time to mess with it. Maybe we should record the need for HTML/Jinja formatting as an issue, what do you think?

**Review by MattF-NSIDC — APPROVED** [review]
